### PR TITLE
support expanding destination on transfers

### PR DESCRIPTION
**Pull request by MDrollette**
This adds support for expanding the polymorphic `destination` property of `Transfer`s.

Addresses #193

### CODE REVIEW

File: transfer.go [hunk 1]
```diff
@@ -20,6 +20,34 @@ type TransferSourceType string
...
+	// which is a bank account.
+	TransferDestinationBankAccount TransferDestinationType = "bank_account"
+
+	// TransferDestinationCard is a constant representing a transfer destination
+	// which is a card.
+	TransferDestinationCard TransferDestinationType = "card"
+
+	// TransferDestinationAccount is a constant representing a transfer destination
+	// which is a Stripe account.
+	TransferDestinationAccount TransferDestinationType = "account"
```
Comment (brandur) [inline transfer.go]: Tiny nit, but mind just moving this up above bank account so we can have it alphabetized?

File: transfer.go [hunk 2]
```diff
@@ -20,6 +20,34 @@ type TransferSourceType string
...
+
+// TransferDestination describes the destination of a Transfer.
+// The Type should indicate which object is fleshed out (eg. BankAccount or Card)
+// For more details see https://stripe.com/docs/api/go#transfer_object
+type TransferDestination struct {
+	Type        TransferDestinationType `json:"object"`
+	ID          string                  `json:"id"`
+	Card        *Card                   `json:"-"`
+	BankAccount *BankAccount            `json:"-"`
+	Account     *Account                `json:"-"`
```
Comment (brandur) [inline transfer.go]: We need to get a real convention for ordering here, but for now can you just alphabetize this list?

File: transfer.go [hunk 3]
```diff
@@ -20,6 +20,34 @@ type TransferSourceType string
...
+	// which is a card.
+	TransferDestinationCard TransferDestinationType = "card"
+
+	// TransferDestinationAccount is a constant representing a transfer destination
+	// which is a Stripe account.
+	TransferDestinationAccount TransferDestinationType = "account"
+)
+
+// TransferDestination describes the destination of a Transfer.
+// The Type should indicate which object is fleshed out (eg. BankAccount or Card)
```
Comment (brandur) [inline transfer.go]: I think a copy + paste error: maybe just add "Account" to this as well.

**Comment by brandur** [comment]
Incredible work here @MDrollette!

Thanks for taking the time to put this together; it's certainly very much a non-trivial change. Great call following the convention for expansion established for payment source types.

I left a couple very minor comments above, but this looks great to me.

**Comment by MDrollette** [comment]
@brandur reordered.

**Comment by brandur** [comment]
LGTM.

**Comment by brandur** [comment]
Released as 17.0.0.

**Pull request merged**